l10n_th_partner/views/res_users_view.xml

```diff
@@ -11,5 +11,13 @@
             </xpath>
         </field>
     </record>
-
+        <record id="view_partner_form_firstname" model="ir.ui.view">
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="partner_firstname.view_partner_form_firstname" />
+        <field name="arch" type="xml">
+            <field name="firstname" position="after">
+                <field name="lastname" position="move"/>
+            </field>
+        </field>
+    </record>
```

Comment:
This is user file not partner file, you can move this record to res_partner_view.xml